src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -1130,6 +1130,8 @@ protected void _renameUsing(Map<String, POJOPropertyBuilder> propMap,
             String rename = null;
             // [databind#4302] since 2.17, Need to skip renaming for Enum properties
             if (!prop.hasSetter() && prop.getPrimaryType().isEnumType()) {
+                // Just put back
+                propMap.put(fullName.getSimpleName(), prop);
```

Comment:
Just a temporary solution to show that not, we should not just skip renaming, but also put back the property...